crnlib/lzma_Types.h

```diff
@@ -90,9 +90,9 @@ typedef int Bool;
 #define True 1
 #define False 0
 
-#ifdef _MSC_VER
+#if defined(_WIN32)
 
-#if _MSC_VER >= 1300
+#if defined(_MSC_VER) && _MSC_VER >= 1300

```

Comment:
I'd leave this part as-is. Mingw does accept these MSVC extensions, but they look like some optimization for very old compilers (which only does anything on x86) that's probably not a good idea now anyway.

Reply:
We need to run this code on 32-bit MinGW build:

```
#define MY_CDECL __cdecl
#define MY_STD_CALL __stdcall
#define MY_FAST_CALL MY_NO_INLINE __fastcall
```

Otherwise some other code elsewhere produce a type mismatch.

Reply:
```
crunch/crnlib/crn_lzma_codec.cpp:12:19: error: invalid conversion
 from ‘int (*)(unsigned char*, size_t*, const unsigned char*, size_t, unsigned char*, size_t*, int, unsigned int, int, int, int, int, int)’ {aka ‘int (*)(unsigned char*, unsigned int*, const unsigned char*, unsigned int, unsigned char*, unsigned int*, int, unsigned int, int, int, int, int, int)’}
 to ‘crnlib::lzma_codec::LzmaCompressFuncPtr’ {aka ‘int (__attribute__((stdcall)) *)(unsigned char*, unsigned int*, const unsigned char*, unsigned int, unsigned char*, unsigned int*, int, unsigned int, int, int, int, int, int)’} [-fpermissive]
   12 |     : m_pCompress(LzmaCompress),
      |                   ^~~~~~~~~~~~
      |                   |
      |                   int (*)(unsigned char*, size_t*, const unsigned char*, size_t, unsigned char*, size_t*, int, unsigned int, int, int, int, int, int) {aka int (*)(unsigned char*, unsigned int*, const unsigned char*, unsigned int, unsigned char*, unsigned int*, int, unsigned int, int, int, int, int, int)}
```